syncstorage/tests/functional/test_storage.py

```diff
@@ -73,27 +73,37 @@ def _switch_user(self):
         finally:
             self.root = orig_root
 
+    def retry(self, method, *args, **kwargs):
+        func = getattr(self.app, method)
+        try:
+            return func(*args, **kwargs)
+        except webtest.AppError as ex:
+            if "500 " not in ex.message and "503 " not in ex.message:
```

Comment:
Oh and one more: we should only retry on 503s. The current issue w/ syncstorage-rs triggering 500s in some conflicting lock_for_write cases is a bug

Reply:
Ok. I'm not a super fan of scraping out the error code from the message like this. I may try something cleaner. 

I'll agree that the `500` is there mostly because the rust is currently returning the wrong error. I'm hoping to have that fixed up separately. I may just keep this branch a WIP until it is.

Reply:
Unfortunately webtest.AppError somewhat swallows responses, the str match is the only way to check the error code on it

With enough effort you could probably pass along `expect_errors=True`, getting back a response object instead, then [manually check for bad status codes](https://github.com/Pylons/webtest/blob/master/webtest/app.py#L663), reraising the AppError yourself. Not sure it's worth bothering